subprojects/launcher/src/main/java/org/gradle/launcher/daemon/configuration/BuildProcess.java

```diff
@@ -52,4 +54,16 @@ public boolean configureForBuild(DaemonParameters requiredBuildParameters) {
         }
         return false;
     }
+
+    /**
+     * Checks whether the current process is using the default client VM setting of 64m, which is too low to run the majority of builds.
+     */
+    private boolean isLowDefaultMemory(List<String> effectiveSingleUseJvmArgs) {
+        for (String arg : effectiveSingleUseJvmArgs) {
+            if (arg.startsWith("-Xmx")) {
+                return false;
+            }
+        }
+        return "64m".equals(getJvmOptions().getMaxHeapSize());
```

Comment:
Can we extract a constant for the `64m` and reference it from both places?

Reply:
Yes we can, though the other place is in buildSrc, so I can't do it in this PR - It first needs a wrapper update.